### PR TITLE
fix: 7.0 support

### PR DESCRIPTION
**Pull request by farfromrefug**
Move to esm to support 7.0
All pieces need to be updated!

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,20 +1,34 @@
...
-        "module": "commonjs",
-        "target": "es5",
-        "lib": [
-            "es6",
-            "dom",
-            "es2015.iterable"
-        ],
-        "noImplicitUseStrict": true,
+        "target": "es2017",
+        "module": "esnext",
```
Comment (PeterStaev) [inline tsconfig.json:4]: Compile options are wrong and creates problems. See https://github.com/NativeScript/nativescript-cli/issues/5396

**Review by PeterStaev — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 1 above)
Comment (farfromrefug) [reply tsconfig.json:4]: They need to be this. Just the hooks need to be build differently.
here is an example https://github.com/nativescript-community/l/blob/master/package.json#L47

Comment (NathanWalker) [reply tsconfig.json:4]: @farfromrefug is this in operational form in it's current state or are there other stack related items that must be updated that you see? Curious if you've manually packed these changes and seen unit testing restored?

Comment (farfromrefug) [reply tsconfig.json:4]: @NathanaelA no it needs to be done. The hook wont work if not es2015. And the unit test project itself needs to be build in es2017

**Comment by NathanWalker** [comment]
Thanks @farfromrefug - @edusperoni was able to assist with the final piece and got unit testing support restored with 1.0.2 from this pr: https://github.com/NativeScript/nativescript-unit-test-runner/pull/44 👍

**Pull request closed without merging**